src/App.tsx

```diff
@@ -1,34 +1,125 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import './App.scss';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
-export const App: React.FC = () => {
+enum SortType {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setViewTodos] = useState<Todo[]>(todos);
+  const [userId, setUserId] = useState<number>(0);
+  const [selectedTodo, setSelectedTodo] = useState<number>(0);
+  const [loading, setLoading] = useState(false);
+  const [filteredBy, setFilterBy] = useState<string>(SortType.ALL);
+  const [query, setQuery] = useState<string>('');
+  const [selectedUser, isUserSelected] = useState<boolean>(false);
```

Comment:
no need to indicate generics for primitives here
```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
  const [visibleTodos, setViewTodos] = useState<Todo[]>(todos);
  const [userId, setUserId] = useState(0);
  const [selectedTodo, setSelectedTodo] = useState(0);
  const [loading, setLoading] = useState(false);
  const [filteredBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
  const [query, setQuery] = useState('');
  const [selectedUser, isUserSelected] = useState(false);
```

Reply:
also, give proper names – you apply filtering, not sorting